### ISSUE TITLE
default-authentication-flow not allowing 2FA Setup Step

### ISSUE THREAD

**Issue by adamzvolanek** [goauthentik/authentik#6173]
**Describe the bug**

Users sent a recovery email link to update/reset password who did not previously setup 2FA cannot setup 2FA after successful sign-in.

**To Reproduce**
1. Have user already created in Authentik
2. Send recovery flow email
3. Update password as user
4. Sent to login screen, and attempt sign-in
5. Prompt appears to setup TOTP or WebAuthn
6. Select either and see screenshot:

![image](https://github.com/goauthentik/authentik/assets/80661527/c77953c9-0d44-4f9b-8ee8-6d3fea830de5)

**Expected behavior**

I expect the setup screen/flow to be presented for users that are valid/able to sign-in GIVEN they do not have TOTP/WebAuth already setup/configured.

**Screenshots**

Default-authentication-login flow below:

![image](https://github.com/goauthentik/authentik/assets/80661527/49b74934-03f0-4bb0-b367-2eb78cff891b)

Authentik Error

![image](https://github.com/goauthentik/authentik/assets/80661527/bac0833f-7513-4ea3-8e9b-ee5a253075ad)

**Logs**
<details>
    <summary>Stacktrace from authentik</summary>

```
Traceback (most recent call last):
  File "/authentik/flows/views/executor.py", line 342, in post
    stage_response = self.current_stage_view.post(request, *args, **kwargs)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/authentik/stages/authenticator_validate/stage.py", line 284, in post
    stage_pk = self.executor.plan.context(PLAN_CONTEXT_SELECTED_STAGE)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
builtins.TypeError: 'dict' object is not callable
```
</details>

**Version and Deployment (please complete the following information):**
- authentik version: 2023.5.4
- Deployment: [e.g. docker-compose, helm]

**Comment by adamzvolanek** [comment]
I created an invitation via enrollment link. The user is able to successfully create an username, password, and subsequently login. However after a logout and attempt to log back in, the bug re-appears and user cannot complete TOTP enrollment.

Image from Authentik logs

![image](https://github.com/goauthentik/authentik/assets/80661527/f3dd0465-268c-4eea-a69d-ed9923c1214e)

**Comment by haupas** [comment]
We experienced the same behaviour after the update to 2023.5.4.

Downgrading back to 2023.5.3 solved the problem for us (at least for now)

**Comment by BeryJu** [comment]
this is fixed by #6062 and will be included in 2023.6
